src/app/grid/grid-drop-indicator/grid-drop-indicator.ts

```diff
@@ -0,0 +1,201 @@
+import { Component, ViewChild, AfterViewInit, OnDestroy, Renderer2 } from '@angular/core';
+import { IDragMoveEventArgs, IgxGridComponent, IRowDragStartEventArgs, Point } from 'igniteui-angular';
+import { DATA } from '../../data/customers';
+import { Subject, interval, Observable, Subscription } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+
+@Component({
+    selector: 'app-grid-drop-indicator',
+    styleUrls: ['grid-drop-indicator.scss'],
+    templateUrl: 'grid-drop-indicator.html'
+})
+export class GridDropIndicatorComponent implements AfterViewInit, OnDestroy {
+  @ViewChild('grid', { read: IgxGridComponent, static: true })
+  public grid: IgxGridComponent;
+  public data: any[];
+  private destroy$ = new Subject();
+  private intervalSub: Subscription;
+  private interval$: Observable<number>;
+  private _draggedRow: any;
+  private highlightedRow: HTMLElement;
+
+  constructor(private renderer: Renderer2) {
+    this.data = DATA;
+  }
+
+  public onDropAllowed(args): void {
```

Comment:
This can be typed (will also need the event interface import)
```suggestion
  public onDropAllowed(args: IDropDroppedEventArgs): void {
```